internal/olm/operator/catalog_source.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package olm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	olmclient "github.com/operator-framework/operator-sdk/internal/olm/client"
+	internalregistry "github.com/operator-framework/operator-sdk/internal/olm/operator/internal"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// CatalogResources holds resources for handling CatalogSource creation
+type CatalogResources struct {
+	namespace       string
+	registryPodName string
+	registryPodAddr string
+
+	catalogsourceObj []runtime.Object
+}
+
+// initialize will initialize catalog resources with values from
+// a registry pod and validate the CatalogResources struct
+func initialize(rp *internalregistry.RegistryPod) (*CatalogResources, error) {
+	if rp == nil {
+		return nil, fmt.Errorf("registry pod is not initialized")
+	}
```

Comment:
```suggestion
	if rp == nil || rp.pod == nil {
		return nil, fmt.Errorf("Registry Pod is not created")
	}
```

Reply:
pod is an exported field in RegistryPod struct, IMO it should be unexported. That's the reason I added a function to get the pod name.

Reply:
I am not sure if I follow your comment. Is not Pod an attribute of the object rp used here?

Reply:
It is an attribute on RegistryPod struct. However, it was unexported. 

I just changed to an exported field to be able to use it here.